test/unified_format.py

```diff
@@ -544,6 +544,19 @@ def setUpClass(cls):
             raise unittest.SkipTest(
                 '%s runOnRequirements not satisfied' % (cls.__name__,))
 
+        # add any special-casing for skipping tests here
+        if client_context.storage_engine == 'mmapv1':
+            if cls.TEST_SPEC['description'].find('retryable-writes') != -1:
+                raise unittest.SkipTest(
+                    "MMAPv1 does not support retryWrites=True")
+            if cls.TEST_SPEC['description'].find('change-streams') != -1:
+                raise unittest.SkipTest(
+                    "MMAPv1 does not support change streams")
+            if cls.TEST_SPEC['description'].find(
+                    'transactions-convenient-api') != -1:
+                raise unittest.SkipTest(
+                    "MMAPv1 does not support document-level locking")
```

Comment:
For these would it make more sense to skip any test that uses startTransaction()/withTransaction()/watch() on mmap rather than using strings (which can change)?

Reply:
Done